### ISSUE TITLE
Accessibility: Unable to access the table header sort control using keyboard navigation

### ISSUE THREAD

**Issue by yousoph** [apache/superset#26176]
#### How to reproduce the bug

1. Create a table chart type 
2. Add the table chart to a dashboard 
3. Using keyboard navigation, tab through the dashboard elements

### Expected results
User is able to navigate to and use the sort controls in table chart types using keyboard navigation
https://www.w3.org/TR/WCAG22/#keyboard

### Actual results
Keyboard navigation skips the table chart headers

#### Screenshots
![image](https://github.com/apache/superset/assets/10627051/ad0ea5cd-6097-45e6-b567-6054e700f332)

### Environment

(please complete the following information):

- browser type and version:
- superset version: `master`
- python version: `python --version`
- node.js version: `node -v`
- any feature flags active:

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by arunthirumani** [comment]
@yousoph @eschutho 

As per my analysis,
1)we need to add tabIndex attribute to <th> in TableChart.tsx
2)we need a keyboard event handler to trigger column sorting when 'Enter' key is pressed on column

can you please assign this to me?  I want to fix this issue.

**Comment by eschutho** [comment]
Thanks for picking up this ticket @arunthirumani! I have a [draft PR](https://github.com/apache/superset/pull/26138/files#diff-0bfb3304e2c6c06116fbc71ed6844bf745ffbae184e64ee777af224f6df483f9R595) of an accessible keyboard event trigger in dashboards if you want it for reference.

**Comment by arunthirumani** [comment]
@eschutho 
Thank you for the reference, it was helpful.
Here is my draft [PR](https://github.com/apache/superset/pull/26326)
please let me know comments/suggestions
